### PR TITLE
Symfony 4 Compatibility

### PR DESCRIPTION
**Pull request by ilohpro**

### CODE REVIEW

**Comment by cristianoc72** [comment]
Please, could you re-run travis? After merging https://github.com/propelorm/Propel2/pull/1434 test suite should be green.
Thank you!

File: Resources/config/propel.xml [hunk 1]
```diff
@@ -35,30 +35,30 @@
...
             <argument>%kernel.root_dir%</argument>
             <argument>%propel.configuration%</argument>
         </service>
 
-        <service id="propel.loader.yaml" class="%propel.loader.yaml.class%">
+        <service id="propel.loader.yaml" class="%propel.loader.yaml.class%" public="true">
             <argument>%kernel.root_dir%</argument>
             <argument>%propel.configuration%</argument>
-            <argument type="service" id="faker.generator" on-invalid="null" />
+            <argument type="service" id="faker.generator" on-invalid="null" public="true"/>
```
Comment (gharlan) [inline Resources/config/propel.xml:58]: I guess the `public="true"` is wrong here (`argument`s don't have a `public` attribute).

**Review by gharlan — CHANGES_REQUESTED** [review]

**Comment by ilohpro** [comment]
My last commit is green, I left you see if it's usefull to merge it to the master...

**Comment by cristianoc72** [comment]
Thank you to include my branch 4.0 into your PR!
Anyway, I'm asking you (and the PropelBundle team) to consider the following points:
1. I refactored the bundle to be Symfony 4.0 only, so it's better to create a new `4.0` branch in Propel repository, instead of merging it into 3.0, because it beaks compatibility with Symfony 3.
2. My 4.0 branch is not finished yet: I have to test the commands behavior with the new directory structure and some configuration properties should be moved from configuration file to `.ENV` file. I plan to do this in a couple of days.
3. Since Symfony 4 directory structure is different from previous versions, there are some changes in PropelBundle 4.0 branch that should be documented. I plan to write a new recipe for Propel cookbook at the end of the next week.
4. Last but not the least, I'm working on a Symfony flex recipe for PropelBundle, so that we could install PropelBundle and Propel simply with `composer require propel`. 

Please, have a little patience again 😄

**Comment by kingchills** [comment]
I too have been working on a [4.0 branch](https://github.com/kingchills/PropelBundle/tree/4.0) for this.  Please feel free to use any of my work on this.  I have implemented registering of the commands, and have tested most of them.  My unit tests all come back green.  One of my concerns of your current work is setting all of the services to be public, which isn't considered best practices within symfony.  Also your changes in Command/AbstractCommand.php are unnecessary if files are placed in the proper place (schemas will be found and moved to the cache if placed in src/propel/*schema.xml).

File: composer.json [hunk 2]
```diff
@@ -16,14 +16,19 @@
...
-        "symfony/dependency-injection": "^2.8|^3.0",
-        "symfony/framework-bundle": "^2.8|^3.0",
-        "symfony/security-acl": "^2.8|^3.0"
+        "symfony/console": "^2.8|^3.0|^4.0",
+        "symfony/dependency-injection": "^2.8|^3.0|^4.0",
+        "symfony/framework-bundle": "^2.8|^3.0|^4.0",
+        "symfony/security-acl": "^2.8|^3.0|^4.0"
+    },
+    "suggest": {
+        "symfony/security-acl": "For using acl instead of voters"
```
Comment (kingchills) [inline composer.json:25]: Does this need to be in the "suggest" if it is in the require?

File: composer.json [hunk 3]
```diff
@@ -16,14 +16,19 @@
...
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8.21|^5.0.10",
-        "sensio/framework-extra-bundle": "^3.0.2",
-        "fzaninotto/faker": "^1.5"
+        "phpunit/phpunit": "^6.0",
+        "sensio/framework-extra-bundle": "^4.0",
+        "fzaninotto/faker": "^1.5",
+        "symfony/security-acl": "^3.0",
```
Comment (kingchills) [inline composer.json:31]: Does not need to be in "require-dev" when in require

**Comment by cristianoc72** [comment]
@kingchills your work is almost perfect, and it's compatible with Symfony 3.4, too! Why don't you submit a PR? Imho, your 4.0 branch is already mergeable, as is.

**Comment by kingchills** [comment]
@cristianoc72  Consider it done.

**Comment by dereuromark** [comment]
Shall we continue here?

**Comment by ilohpro** [comment]
It's quite old, I think this pull request has to been fully reviewed on new versions.
Personnally, i don't use it anymore, maybe i should close this PR too.

**Comment by dereuromark** [comment]
Actually, https://github.com/SkyFoxvn/PropelBundle will most likely be the follower of this now.

**Pull request closed without merging**